src/interface/spy.jl

```diff
@@ -100,7 +102,23 @@ See also
 [`BrailleCanvas`](@ref), [`BlockCanvas`](@ref),
 [`AsciiCanvas`](@ref), [`DotCanvas`](@ref)
 """
-spy(A::AbstractMatrix; kwargs...) = spy(size(A)..., _findnz(A)...; kwargs...)
+
+function spy(A::AbstractMatrix; kwargs...)
+    rows, cols, vals = _strict_non_zeros(_findnz(A)...)
+    if get(kwargs, :zeros, false)
+        I = CartesianIndex.(zip(rows, cols))  # non zeros
```

Comment:
How about

```julia
rows, cols, vals = _findnz(A)
nonzero_R= findall(!iszero, vals)
if get(kwargs, :zeros, false)
    I = [CartesianIndex(rows[idx], cols[idx]) for idx in nonzero_R]
    mask = trues(size(A))
    mask[I] .= false;
    Z = CartesianIndices(axes(A))[mask]
    ...
end
```

Reply:
Damn I really feel stupid on this one :sweat_smile:, thanks !

Reply:
The idea of `I = [CartesianIndex(rows[idx], cols[idx]) for idx in nonzero_R]` is that you don't need to call `rows[I]` to allocate two memory blocks for `Array{Int}`; thus it can saves two memory allocations.

Reply:
You still need `rows[I]` and `cols[I]` with zeros eliminated in the call to `spy(size(A)..., rows, cols, vals; kwargs...)`, so I don't think we can avoid new allocation here ?